mne/viz/utils.py

```diff
@@ -167,43 +167,57 @@ def mne_analyze_colormap(limits=[5, 10, 15], format='mayavi'):
         brain.scale_data_colormap(fmin=-15, fmid=0, fmax=15, transparent=False)
 
     """
-    l = np.asarray(limits, dtype='float')
-    if len(l) != 3:
-        raise ValueError('limits must have 3 elements')
-    if any(l < 0):
-        raise ValueError('limits must all be positive')
-    if any(np.diff(l) <= 0):
+    # Ensure limits is an array
+    limits = np.asarray(limits, dtype='float')
+
+    if len(limits) != 3 and len(limits) != 6:
+        raise ValueError('limits must have 3 or 6 elements')
+    if len(limits) == 3 and any(limits < 0.):
+        raise ValueError('if 3 elements, limits must all be non-negative')
+    if any(np.diff(limits) <= 0):
```

Comment:
shouldn't the limits be monotonically increasing only if the number of elements is 3?